src/package-tree.ts

```diff
@@ -1,53 +1,116 @@
+
+    // Step 2: Look at package-lock.json to find dependendencies
+    const result =  await getPackageTreeFromDependencyList(dependencies, pjsonLockObj);
+
+    const treeHead: PackageTree<undefined> = {"rootPackageName": pjsonObj.name,
+      "version": pjsonObj.version,
+      "data": undefined,
+      "dependencies": result}; 
+    
+      return treeHead;
+  }catch(error){
+    throw error;
+  }
+
 }
 
-function getPackagePOIList(package: PackageTree): PointOfInterest[] {
-  // TODO:
-  // calls getPointsOfInterest for each file in package
-  throw new Error('not implemented');
+/**
+ * Takes in a list of depndencies and returns a constructed PackageTree
+ * @param packageLock the package-lock.json file of the root project
+ */
+export async function getPackageTreeFromDependencyList(dependencies: {}, packageLock: {}): Promise<Array<PackageTree<undefined>>|undefined>{
+// Step 1: For each dependency create a PackageTree obj with the name and version fields populated
+if(dependencies){
+  return undefined;
 }
+const dependencyArr: Array<[string,string]> = Object.entries(dependencies);
+const packageTreeArr: Array<PackageTree<undefined>> = [];
+ dependencyArr.forEach((element: [string, string]) => {
+   const pkgTreeObj: PackageTree<undefined> = { "rootPackageName": element[0],
+                                                "version": element[1],
+                                                "data": undefined,
+                                                "dependencies": []
+                                                }; 
+                                                
+    packageTreeArr.push(pkgTreeObj);
+ });
 
-// Gets Points of Interest for a single file
-function getPointsOfInterest(
-    contents: string, fileName: string,
-    functionArray: POIDetector[]): PointOfInterest[] {
-  // TODO:
-  throw new Error('not implemented');
+// Step 2: For each Package Tree obj get the dependencies
+const m: Array<Promise<void>> = packageTreeArr.map(async element => {
+  element.dependencies = await populateDependencies(element, packageLock);
+});
+await Promise.all(m);
+return packageTreeArr.sort(compare);
+}
 
-  // const pointsOfInterest: PointOfInterest[] = [];
-  // functionArray.forEach((f)=> pointsOfInterest.push(...f(contents,
-  // fileName))); return pointsOfInterest;
+/**
+ * Takes in a PackageTree and populates its dependency field
+ * Currently using 'any' type for pjsonLock because the types for package-lock.json have not been written yet
+ */
+async function populateDependencies(pkg: PackageTree<undefined>, packageLock: any): Promise<Array<PackageTree<undefined>>|undefined>{
+ const packageName = pkg.rootPackageName;
+ const dependencies = packageLock.dependencies[packageName].requires;
+ if(!dependencies){
+   return undefined;
+ }
+ 
+ const dependencyPackageTree = await getPackageTreeFromDependencyList(dependencies, packageLock);
+ if(dependencyPackageTree){
+   return dependencyPackageTree;
+ }
+ return undefined;
```

Comment:
Same note here concerning [] vs undefined. You can shorthand the last four lines as `return dependencyPackageTree || undefined`.